### PR TITLE
fix: ensure Dropdown aria-labelledby to have an ID #11572

### LINKED ISSUE #11572: Dropdowns have broken ARIA  labelledby
## What should happen?

aria-labelledby must point to label in dropdowns.

## What happens instead?

It's creating labelledby attribute but without point to the label.

## Possible Solution

Add id to this.$currentAnchor when created with GetYoDigits function.

## Test Case and/or Steps to Reproduce (for bugs)
Go to https://foundation.zurb.com/sites/docs/dropdown.html
Inspect dropdowns, they have aria-labelledby attribute but buttons doesn't have this id.

## Your Environment
Affects current foundation version 6.5

## Checklist
- [x] I have read and follow the CONTRIBUTING.md document.
- [x] There are no other issues similar to this one.
- [x] The issue title and template are correctly filled.

### PR DESCRIPTION
**Pull request by ncoden**
Closes https://github.com/zurb/foundation-sites/issues/11572

## Description

Currently, Dropdown generate a random id for `aria-labelledby` if it is not set to target the anchor. But the anchor id is never set.

This pull request fix this by setting the anchor id with the generated one if the anchor doens't have an ID already.

Closes https://github.com/zurb/foundation-sites/issues/11572

## Types of changes

- [ ] Documentation
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (anything that would change an existing functionality)
- [ ] Maintenance (refactor, code cleaning, development tools...)

## Checklist

- [x] I have read and follow the CONTRIBUTING.md document.
- [x] The pull request title and template are correctly filled.
- [x] The pull request targets the right branch (`develop` or `develop-v...`).
- [x] My commits are correctly titled and contain all relevant information.
- ~~I have updated the documentation accordingly to my changes (if relevant).~~
- ~~I have added tests to cover my changes (if relevant).~~

### CODE REVIEW

File: js/foundation.dropdown.js [hunk 1]
```diff
@@ -68,16 +68,20 @@ class Dropdown extends Positionable {
       this.$parent = null;
     }
 
-    // Do not change the `labelledby` if it is defined
-    var labelledby = this.$element.attr('aria-labelledby')
-      || this.$currentAnchor.attr('id')
-      || GetYoDigits(6, 'dd-anchor');
+    // Set [aria-describedby] on the Dropdown if it is not set
+    if (typeof this.$element.attr('aria-describedby') === 'undefined') {
```
Comment (silenceway) [inline js/foundation.dropdown.js]: labelledby.
```suggestion
    if (typeof this.$element.attr('aria-labelledby') === 'undefined') {
```

**Review by silenceway — COMMENTED** [review]
Attribute name must be changed.

File: js/foundation.dropdown.js (hunk 1 above)
Comment (ncoden) [reply js/foundation.dropdown.js]: What a stupid mistake. Thank you.

**Review by silenceway — APPROVED** [review]

**Comment by ncoden** [comment]
Note: I checked all others usages of `GetYoDigits()`.

**Comment by joeworkman** [comment]
This pull request has been mentioned on **Foundation Open Source Community**. There might be relevant details there:

https://foundation.discourse.group/t/foundation-for-sites-v6-6-0-is-here-farout/30/1

**Pull request merged**